task7/Working with posts.js

```diff
@@ -0,0 +1,353 @@
+	if (arrayPosts.length == 0) {
+		document.write("No any posts");
+	}
+	else {
+		arrayPosts.forEach(p=>p.output());
+	}
+}
+
+function printLengthOfArray(arr) {
+	document.write('The length of posts ' + arr.length + '<br\/>');
+}
+
+class PostContainer {
+
+	constructor(posts) {
+		this._posts = posts;
+		this._count = this._posts.length;
+	}
+
+	get posts() {
+		return this._posts;
+	}
+
+	set posts(arr) {
+		this._posts = arr;
+	}
+
+	get count() {
+		return this._count;
+	}
+
+	set count(count) {
+		this._count = count;
+	}
+
+	_liesInInterval(date, dateInterval) {
+		return dateInterval == null || dateInterval.left <= date && date <= dateInterval.right;
+	}
+
+	get liesInInterval() {
+		return this._liesInInterval;
+	}
+
+	_hasAnyTag(hashTagsOfPost, hashTags) {
+		return hashTagsOfPost.some(hashTagOfPost=>(hashTags.includes(hashTagOfPost)));
+	}
+
+	get hasAnyTag() {
+		return this._hasAnyTag;
+	}
+
+	_compDates(x, y) {
+		return x.createdAt - y.createdAt;
+	}
+
+	get compDates() {
+		return this._compDates;
+	}
+
+	_isTag(str) {
+		return str[0] == '#';
+	}
+
+	get isTag() {
+		return this._isTag;
+	}
+
+	_isString(obj) {
+		return typeof obj == 'string';
+	}
+
+	get isString() {
+		return this._isString;
+	}
+
+  _rightDate(date) {
+		return date.getHours() >= 0 && date.getHours() <= 23 &&
+		 date.getMinutes() >= 0 && date.getMinutes() <= 59 &&
+		 	 date.getSeconds() >= 0 && date.getSeconds() <= 59;
+	}
+
+	get rightDate() {
+		return this._rightDate;
+	}
+
+	_getPostsByFilter(filterConfig) {
+		return this.posts.filter(element=>
+			(filterConfig.author == "" || element.author == filterConfig.author) &&
+			((filterConfig.dateInterval == null) || this.liesInInterval(element.createdAt, filterConfig.dateInterval)) &&
+			 ((filterConfig.hashTags == []) || this.hasAnyTag(element.hashTags, filterConfig.hashTags))
+			 );
+	}
+
+	get getPostsByFilter() {
+		return this._getPostsByFilter;
+	}
+
+	getPosts(filterConfig, skip=0, top=10) {
+		if (skip < 0 || top <= 0) {
+			return [];
+		}
+		let postsByFilter = this.getPostsByFilter(filterConfig).sort(this.compDates);
+		return postsByFilter.slice(skip, skip + top);
+	}
+
+	getPost(ID) {
+		return this.posts.find(item=>item.id == ID);
+	}
+
+	validatePost(somePost) {
+		return this.isString(somePost.description)&&
+		 somePost.description.length > 0 && somePost.description.length < 200 &&
+		 this.isString(somePost.author) && somePost.author.length > 0 &&
+		 somePost.hashTags.every(this.isTag) &&
+		 somePost.hashTags.every(this.isString) && somePost.likes.every(this.isString) &&
+		 somePost.createdAt instanceof Date && this.rightDate(somePost.createdAt);
+	}
+
+	addPost(somePost) {
+		if (!this.validatePost(somePost)) {
+			return false;
+		}
+		this.posts.push(somePost);
+		this.count += 1;
+		return true;
+	}
+
+	editPost(ID, edit) {
+		let index = this.posts.findIndex(item=>item.id==ID);
+		if (index==-1) {
+			return false;
+		}
+		let edPost = this.posts[index];
```

Comment:
Object is reference type changing it here changes it in array. So, validation can be failed, but post already changed